core/src/test/java/io/aiven/kafka/tieredstorage/fetch/index/MemorySegmentIndexesCacheTest.java

```diff
@@ -0,0 +1,291 @@
+        }
+
+        @AfterEach
+        void tearDown() {
+            reset(removalListener);
+        }
+
+        @Test
+        void noEviction() throws IOException, StorageBackendException {
+            cache.configure(Map.of(
+                "size", "-1",
+                "retention.ms", "-1"
+            ));
+            assertThat(cache.cache.asMap()).hasSize(0);
+
+            final ObjectKey key = objectKeyFactory.key(REMOTE_LOG_SEGMENT_METADATA, ObjectKeyFactory.Suffix.INDEXES);
+            final InputStream offsetIndex = cache.get(
+                key,
+                IndexType.OFFSET,
+                offsetIndexSupplier
+            );
+            assertThat(offsetIndex).hasBinaryContent(OFFSET_INDEX);
+            assertThat(cache.cache.asMap()).hasSize(1);
+            final InputStream cachedOffsetIndex = cache.get(
+                key,
+                IndexType.OFFSET,
+                offsetIndexSupplier
+            );
+            assertThat(cachedOffsetIndex).hasBinaryContent(OFFSET_INDEX);
+            verifyNoMoreInteractions(offsetIndexSupplier);
+            assertThat(cache.cache.asMap()).hasSize(1);
+
+            final InputStream timeIndex = cache.get(
+                key,
+                IndexType.TIMESTAMP,
+                timeIndexSupplier
+            );
+            assertThat(timeIndex).hasBinaryContent(TIME_INDEX);
+            assertThat(cache.cache.asMap()).hasSize(2);
+            final InputStream cachedTimeIndex = cache.get(
+                key,
+                IndexType.TIMESTAMP,
+                timeIndexSupplier
+            );
+            assertThat(cachedTimeIndex).hasBinaryContent(TIME_INDEX);
+            verifyNoMoreInteractions(timeIndexSupplier);
+            assertThat(cache.cache.asMap()).hasSize(2);
+
+            verifyNoMoreInteractions(removalListener);
+        }
+
+        @Test
+        void timeBasedEviction() throws IOException, StorageBackendException, InterruptedException {
+            cache.configure(Map.of(
+                "size", "-1",
+                "retention.ms", "100"
+            ));
+            assertThat(cache.cache.asMap()).hasSize(0);
+
+            final ObjectKey key = objectKeyFactory.key(REMOTE_LOG_SEGMENT_METADATA, ObjectKeyFactory.Suffix.INDEXES);
+            final InputStream offsetIndex = cache.get(
+                key,
+                IndexType.OFFSET,
+                offsetIndexSupplier
+            );
+            assertThat(offsetIndex).hasBinaryContent(OFFSET_INDEX);
+            assertThat(cache.cache.asMap()).hasSize(1);
+            final InputStream cachedOffsetIndex = cache.get(
+                key,
+                IndexType.OFFSET,
+                offsetIndexSupplier
+            );
+            assertThat(cachedOffsetIndex).hasBinaryContent(OFFSET_INDEX);
+            verifyNoMoreInteractions(offsetIndexSupplier);
+            assertThat(cache.cache.asMap()).hasSize(1);
+
+            Thread.sleep(100);
```

Comment:
It's worth adding a comment that this sleep is needed to spread away in time the creation (and eviction) of the cache entries, or something like this

Reply:
Instead of just waiting, should we wait for removal listener to be called or for size becoming zero?

Reply:
@AnatolyPopov, falling back to using thread sleep (same as in ChunkCacheTest) as adding a stronger condition (e.g. waiting for cache to be empty) takes unknown amount of time github actions--unless there's a better approach here.